pkg/plugin/aggregation/aggregator.go

```diff
@@ -106,6 +124,13 @@ func (a *Aggregator) Wait(stop chan bool) {
 			return
 		}
 	}
+
+	// Give all clients a chance to retry failed requests.
+	for _, failedTime := range a.FailedResults {
+		if time.Now().Sub(failedTime) < a.retryWindow {
```

Comment:
maybe refactor this out as well, something like `shouldWaitForRetryWindow` or something?